mongoengine/connection.py

```diff
@@ -234,6 +238,8 @@ def disconnect(alias=DEFAULT_CONNECTION_NAME):
     if alias in _connection_settings:
         del _connection_settings[alias]
 
+    _clear_session()
```

Comment:
Hmm - I missed this in my update. I'm thinking we'll just want to empty the `_LocalSessions.sessions` property - you're closing a connection, the sessions of that connection won't be of any use to anymore.

Reply:
Then again, given you can disconnect from alias X, but connection Y could still be open. Do you think this now pushes us into the territory of "we need to track sessions across multiple connections"?

Reply:
After thinking about it more and reviewing my original proposal, I do not believe this requires tracking sessions across multiple connections.

Clearing out any remaining sessions should suffice - and I've updated accordingly.

I'm OK **not** supporting someone trying to execute a disconnect of connection X in the middle of a transaction in the context of a session bound to connection Y 

😄